File: goldendb/src/test/java/com/dbx/agent/goldendb/GoldendbAgentMetadataTest.java
package com.dbx.agent.goldendb;

import com.dbx.agent.test.TestSupport;
import com.dbx.agent.test.JdbcAgentFake;
import com.dbx.agent.test.JdbcMetadataSqlFake;
import java.sql.Connection;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class GoldendbAgentMetadataTest {
    @Test
    void quotesSchemaAndTableIdentifiersInIndexMetadataSql() {
        GoldendbAgent agent = new GoldendbAgent();
        Connection fake = JdbcMetadataSqlFake.connection();
        TestSupport.setPrivateConnection(agent, fake);

        agent.listIndexes("bad`schema", "bad`table");

        Assertions.assertEquals(
            List.of("SHOW INDEX FROM `bad``table` FROM `bad``schema`"),
            JdbcMetadataSqlFake.statements
        );
    }
}

File: goldendb/src/test/kotlin/com/dbx/agent/goldendb/GoldendbAgentMetadataTest.kt
package com.dbx.agent.goldendb

import com.dbx.agent.test.JdbcMetadataSqlFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class GoldendbAgentMetadataTest {
    @Test
    fun `quotes schema and table identifiers in index metadata SQL`() {
        val agent = GoldendbAgent()
        val fake = JdbcMetadataSqlFake.connection()
        setPrivateConnection(agent, fake)

        agent.listIndexes("bad`schema", "bad`table")

        assertEquals(
            listOf("SHOW INDEX FROM `bad``table` FROM `bad``schema`"),
            JdbcMetadataSqlFake.statements,
        )
    }
}
